BDArmory/UI/BDArmorySetup.cs

```diff
@@ -3467,8 +3467,44 @@ void WindowSettings(int windowID)
                 {
                     CheatOptions.InfiniteElectricity = BDArmorySettings.INFINITE_EC;
                 }
+                if (!BDArmorySettings.G_LIMITS && (HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits || HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits))
+                    BDArmorySettings.G_LIMITS = true; 
+                BDArmorySettings.PART_GLIMIT = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits; // Sync with the Game Difficulty window if the checkbox was toggled there.
+                BDArmorySettings.KERB_GLIMIT = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits;
+                BDArmorySettings.G_TOLERANCE = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().KerbalGToleranceMult * 20.5f;
+                if (BDArmorySettings.G_LIMITS != (BDArmorySettings.G_LIMITS = GUI.Toggle(SLeftRect(line), BDArmorySettings.G_LIMITS, StringUtils.Localize("#LOC_BDArmory_Settings_GLimitsMode"))))//G-Force Limits
+                {
+                    if (!BDArmorySettings.G_LIMITS)
+                    {
+                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits = false;
+                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits = false;
+                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().KerbalGToleranceMult = 1f;
+                        BDArmorySettings.KERB_GLIMIT = false;
+                        BDArmorySettings.PART_GLIMIT = false;
+                    }
+                }
+                if (BDArmorySettings.G_LIMITS)
+                {
+                    if (BDArmorySettings.PART_GLIMIT != (BDArmorySettings.PART_GLIMIT = GUI.Toggle(SLeftRect(++line, 1), BDArmorySettings.PART_GLIMIT, StringUtils.Localize("#autoLOC_140950"))))//Part G-Force Limits
```

Comment:
Setting the part and kerb g-limits then restarting KSP, it doesn't look like these are being applied to the current game's parameters (or at least, not saved) as they're being reset on the next load of KSP. I'm not sure why yet though.

Reply:
OK, it looks like they're being applied, just not saved. If you hit the "Apply" button in KSP's settings window, then they get saved, so it looks like the `AdvancedParams` instance needs saving to a `ConfigNode` somehow.

Reply:
OK, I found out that triggering a save will get those settings to be persistent. I've set the `advancedParams` instance to get refreshed whenever the BDA settings window gets opened, which seems to be enough.
```diff
diff --git a/BDArmory/Competition/BDATournament.cs b/BDArmory/Competition/BDATournament.cs
index 39040f7ee..a0c757989 100644
--- a/BDArmory/Competition/BDATournament.cs
+++ b/BDArmory/Competition/BDATournament.cs
@@ -2128,3 +2128,3 @@ bool GenerateCleanGame()
             }
-            // Update the game state and save it to the persistent save (sine that's what eventually ends up getting loaded when we call Start()).
+            // Update the game state and save it to the persistent save (since that's what eventually ends up getting loaded when we call Start()).
             HighLogic.CurrentGame.Updated();
diff --git a/BDArmory/UI/BDArmorySetup.cs b/BDArmory/UI/BDArmorySetup.cs
index 7dce0253b..9e3034cc3 100644
--- a/BDArmory/UI/BDArmorySetup.cs
+++ b/BDArmory/UI/BDArmorySetup.cs
@@ -220,2 +220,4 @@ public static float SystemMaxMemory
 
+        static GameParameters.AdvancedParams advancedParams;
+
         //competition mode
@@ -646,3 +648,3 @@ void ToggleToolbarButton()
             if (HighLogic.LoadedSceneIsFlight) { windowBDAToolBarEnabled = !windowBDAToolBarEnabled; }
-            else { windowSettingsEnabled = !windowSettingsEnabled; }
+            else { ToggleWindowSettings(); }
         }
@@ -710,2 +712,3 @@ void ToggleWindowSettings()
                 // LoadConfig(); // Don't reload settings, since they're already loaded and mess with other settings windows.
+                if (HighLogic.CurrentGame != null) advancedParams = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>(); // Grab the current AdvancedParams.
             }
@@ -3469,7 +3472,8 @@ void WindowSettings(int windowID)
                 }
-                if (!BDArmorySettings.G_LIMITS && (HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits || HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits))
+                bool advParamsChanged = false;
+                if (!BDArmorySettings.G_LIMITS && (advancedParams.GPartLimits || advancedParams.GKerbalLimits))
                     BDArmorySettings.G_LIMITS = true;
-                BDArmorySettings.PART_GLIMIT = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits; // Sync with the Game Difficulty window if the checkbox was toggled there.
-                BDArmorySettings.KERB_GLIMIT = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits;
-                BDArmorySettings.G_TOLERANCE = HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().KerbalGToleranceMult * 20.5f;
+                BDArmorySettings.PART_GLIMIT = advancedParams.GPartLimits; // Sync with the Game Difficulty window if the checkbox was toggled there.
+                BDArmorySettings.KERB_GLIMIT = advancedParams.GKerbalLimits;
+                BDArmorySettings.G_TOLERANCE = advancedParams.KerbalGToleranceMult * 20.5f;
                 if (BDArmorySettings.G_LIMITS != (BDArmorySettings.G_LIMITS = GUI.Toggle(SLeftRect(line), BDArmorySettings.G_LIMITS, StringUtils.Localize("#LOC_BDArmory_Settings_GLimitsMode"))))//G-Force Limits
@@ -3478,5 +3482,6 @@ void WindowSettings(int windowID)
                     {
-                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits = false;
-                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits = false;
-                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().KerbalGToleranceMult = 1f;
+                        advancedParams.GPartLimits = false;
+                        advancedParams.GKerbalLimits = false;
+                        advancedParams.KerbalGToleranceMult = 1f;
+                        advParamsChanged = true;
                         BDArmorySettings.KERB_GLIMIT = false;
@@ -3489,3 +3494,4 @@ void WindowSettings(int windowID)
                     {
-                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GPartLimits = BDArmorySettings.PART_GLIMIT;
+                        advancedParams.GPartLimits = BDArmorySettings.PART_GLIMIT;
+                        advParamsChanged = true;
                     }
@@ -3493,3 +3499,4 @@ void WindowSettings(int windowID)
                     {
-                        HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().GKerbalLimits = BDArmorySettings.KERB_GLIMIT;
+                        advancedParams.GKerbalLimits = BDArmorySettings.KERB_GLIMIT;
+                        advParamsChanged = true;
                     }
@@ -3497,6 +3504,7 @@ void WindowSettings(int windowID)
                     {
-                        GUI.Label(SLeftSliderRect(++line, 1), $"{StringUtils.Localize("#autoLOC_140956")}:  ({BDArmorySettings.G_TOLERANCE})", leftLabel);//Kerbal G-Force Tolerance
-                        if (BDArmorySettings.G_TOLERANCE != (BDArmorySettings.G_TOLERANCE = Mathf.RoundToInt(GUI.HorizontalSlider(SRightSliderRect(line), BDArmorySettings.G_TOLERANCE, 1f, 40f) * 10f) / 10f))
+                        GUI.Label(SLeftSliderRect(++line, 1), $"{StringUtils.Localize("#autoLOC_140956")}:  ({BDArmorySettings.G_TOLERANCE:0.0}g)", leftLabel);//Kerbal G-Force Tolerance
+                        if (BDArmorySettings.G_TOLERANCE != (BDArmorySettings.G_TOLERANCE = BDAMath.RoundToUnit(GUI.HorizontalSlider(SRightSliderRect(line), BDArmorySettings.G_TOLERANCE, 1f, 40f), 0.5f)))
                         {
-                            HighLogic.CurrentGame.Parameters.CustomParams<GameParameters.AdvancedParams>().KerbalGToleranceMult = BDArmorySettings.G_TOLERANCE / 20.5f; //Default 0.5 Courage BadS Pilot kerb has a GLimit of 20.5
+                            advancedParams.KerbalGToleranceMult = BDArmorySettings.G_TOLERANCE / 20.5f; //Default 0.5 Courage BadS Pilot kerb has a GLimit of 20.5
+                            advParamsChanged = true;
                         } //G-Limit is point at which Kerbs begin to fill the G-meter; e.g. a G_TOLERANCE of 0.4 (g-Limit of 8.16) would result in Kerbs indefiniately tolerating a 8G turn, but a 9 G sustained turn would *very* slowly fill the meter,
@@ -3505,2 +3513,6 @@ void WindowSettings(int windowID)
                 }
+                if (advParamsChanged)
+                {
+                    GamePersistence.SaveGame("persistent", HighLogic.SaveFolder, SaveMode.OVERWRITE); // Update the persistent save.
+                }
                 // Resource steal
```